src/Pate/Memory/MemTrace.hs

```diff
@@ -540,23 +544,24 @@ initMemTrace ::
   IO (MemTraceImpl sym ptrW)
 initMemTrace sym Addr32 = do
   arr <- ioFreshConstant sym "InitMem" knownRepr
-  return $ MemTraceImpl mempty arr
+  return $ MemTraceImpl mempty (MemTraceState arr)
 initMemTrace sym Addr64 = do
   arr <- ioFreshConstant sym "InitMem" knownRepr
-  return $ MemTraceImpl mempty arr
+  return $ MemTraceImpl mempty (MemTraceState arr)
 
-initMemTraceVar ::
+
+mkMemoryBinding ::
   forall sym ptrW.
-  IsSymInterface sym =>
-  sym ->
-  AddrWidthRepr ptrW ->
-  IO (MemTraceImpl sym ptrW, MemTraceVar sym ptrW)
-initMemTraceVar sym Addr32 = do
-  arr <- ioFreshConstant sym "InitMem" knownRepr
-  return $ (MemTraceImpl mempty arr, MemTraceVar arr)
-initMemTraceVar sym Addr64 = do
-  arr <- ioFreshConstant sym "InitMem" knownRepr
-  return $ (MemTraceImpl mempty arr, MemTraceVar arr)
+  MemTraceState sym ptrW ->
+  -- ^ initial memory state (appears in the the given expression when the binding is applied)
```

Comment:
If you want to put the arrows at the end of the line, you probably want to introduce this comment before the argument using `-- |`